Content.Server/Blob/BlobbernautSystem.cs

```diff
@@ -0,0 +1,124 @@
+using System.Linq;
+using System.Numerics;
+using Content.Server.Emp;
+using Content.Server.Explosion.EntitySystems;
+using Content.Shared.Blob;
+using Content.Shared.Damage;
+using Content.Shared.Interaction.Events;
+using Content.Shared.Mobs;
+using Content.Shared.Popups;
+using Content.Shared.Weapons.Melee.Events;
+using Robust.Shared.GameStates;
+using Robust.Shared.Map;
+using Robust.Shared.Random;
+using Robust.Shared.Timing;
+
+namespace Content.Server.Blob
+{
+    public sealed class BlobbernautSystem : EntitySystem
+    {
+        [Dependency] private readonly IMapManager _map = default!;
+        [Dependency] private readonly DamageableSystem _damageableSystem = default!;
+        [Dependency] private readonly SharedPopupSystem _popup = default!;
+        [Dependency] private readonly IGameTiming _gameTiming = default!;
+        [Dependency] private readonly ExplosionSystem _explosionSystem = default!;
+        [Dependency] private readonly IRobustRandom _random = default!;
+        [Dependency] private readonly EmpSystem _empSystem = default!;
+
+        public override void Initialize()
+        {
+            base.Initialize();
+
+            SubscribeLocalEvent<BlobbernautComponent, MobStateChangedEvent>(OnMobStateChanged);
+            SubscribeLocalEvent<BlobbernautComponent, ComponentGetState>(OnGetState);
+            SubscribeLocalEvent<BlobbernautComponent, MeleeHitEvent>(OnMeleeHit);
+        }
+
+        private void OnMeleeHit(EntityUid uid, BlobbernautComponent component, MeleeHitEvent args)
+        {
+            if (args.HitEntities.Count >= 1)
+                return;
+            if (!TryComp<BlobTileComponent>(component.Factory, out var blobTileComponent))
+                return;
+            if (!TryComp<BlobCoreComponent>(blobTileComponent.Core, out var blobCoreComponent))
+                return;
+            if (blobCoreComponent.CurrentChem == BlobChemType.ExplosiveLattice)
```

Comment:
This should be a discrete component that does it's own event handling, not a comparison against an enum.